cardano-node/src/Cardano/Node/Protocol/Shelley.hs

```diff
@@ -155,23 +154,43 @@ readLeaderCredentialsSingleton
      } = pure []
 -- Or to supply all of the files
 readLeaderCredentialsSingleton
-   ProtocolFilepaths
-     { shelleyCertFile      = Just certFile,
-       shelleyVRFFile       = Just vrfFile,
-       shelleyKESFile       = Just kesFile
-     } =
-     fmap (:[]) $
-     mkPraosLeaderCredentials
-       <$> firstExceptT FileError (newExceptT $ readFileTextEnvelope AsOperationalCertificate certFile)
-       <*> firstExceptT FileError (newExceptT $ readFileTextEnvelope (AsSigningKey AsVrfKey) vrfFile)
-       <*> firstExceptT FileError (newExceptT $ readFileTextEnvelope (AsSigningKey AsKesKey) kesFile)
+  ProtocolFilepaths { shelleyCertFile = Just opCertFile,
+                      shelleyVRFFile = Just vrfFile,
+                      shelleyKESFile = Just kesFile
+                    } = do
+    vrfSKey <-
+      firstExceptT FileError (newExceptT $ readFileTextEnvelope (AsSigningKey AsVrfKey) vrfFile)
+
+    (opCert, kesSKey) <- kesKeyCheck kesFile opCertFile
+
+    return [mkPraosLeaderCredentials opCert vrfSKey kesSKey]
+
 -- But not OK to supply some of the files without the others.
 readLeaderCredentialsSingleton ProtocolFilepaths {shelleyCertFile = Nothing} =
-     throwError OCertNotSpecified
+     left OCertNotSpecified
 readLeaderCredentialsSingleton ProtocolFilepaths {shelleyVRFFile = Nothing} =
-     throwError VRFKeyNotSpecified
+     left VRFKeyNotSpecified
 readLeaderCredentialsSingleton ProtocolFilepaths {shelleyKESFile = Nothing} =
-     throwError KESKeyNotSpecified
+     left KESKeyNotSpecified
+
+kesKeyCheck
+  :: FilePath
+  -- ^ KES key
+  -> FilePath
+  -- ^ Operational certificate
+  -> ExceptT PraosLeaderCredentialsError IO (OperationalCertificate, SigningKey KesKey)
+kesKeyCheck kesFile certFile = do
```

Comment:
This function appears to be loading the KES key and opcert rather than just check.  Maybe the function name could reflect that.

Reply:
`opCertKesKeyCheck`?